RestaurantReservation.Db/Repositories/OrderRepository.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Repositories.IRepositories;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class OrderRepository : EntityRepository<Order>, IOrderRepository
+    {
+
+        public OrderRepository(RestaurantReservationDbContext dbContext) : base(dbContext)
+        {
+        }
+
+        public async Task<decimal> CalculateAverageOrderAmountAsync(int employeeId)
+        {
+            var OrderAmountList = await _dbContext.Orders.AsNoTracking()
+                       .Where(o => o.EmployeeId == employeeId)
+                       .Include(o => o.OrderItems)
+                       .ThenInclude(oi=> oi.MenuItem)
+                       .AsSplitQuery()
+                       .ToListAsync();
+
+            return OrderAmountList.Count() > 0 ?
+                 OrderAmountList.Select(o => o.TotalAmount).Average() : 0;
+        }
+
+        public async Task<IEnumerable<MenuItem>?> ListOrderedMenuItemsAsync(int reservationId)
+        {
+            return await _dbContext.Orders.AsNoTracking()
```

Comment:
post the generated SQL query.

Reply:
info: 1/8/2024 09:05:17.586 RelationalEventId.CommandExecuted[20101] (Microsoft.EntityFrameworkCore.Database.Command)
      Executed DbCommand (253ms) [Parameters=[@__reservationId_0='9'], CommandType='Text', 
CommandTimeout='30']
      SELECT DISTINCT [t].[MenuItemId], [t].[Description], [t].[Name], [t].[Price], [t].[RestaurantId]
      FROM [Orders] AS [o]
      INNER JOIN (
          SELECT [m].[MenuItemId], [m].[Description], [m].[Name], [m].[Price], [m].[RestaurantId], [o0].[OrderId]
          FROM [OrderItems] AS [o0]
          INNER JOIN [MenuItems] AS [m] ON [o0].[MenuItemId] = [m].[MenuItemId]
      ) AS [t] ON [o].[OrderId] = [t].[OrderId]
      WHERE [o].[ReservationId] = @__reservationId_0